src/csharp/Grpc.Core/Internal/AsyncCallBase.cs

```diff
@@ -157,8 +157,9 @@ protected void StartReadMessageInternal(AsyncCompletionDelegate<TRead> completio
                 Preconditions.CheckNotNull(completionDelegate, "Completion delegate cannot be null");
                 CheckReadingAllowed();
 
-                call.StartReceiveMessage(HandleReadFinished);
+                // I feel like a prerequisite variable should be set before entering that code path
```

Comment:
The return from native which is started by StartReceiveMessage will use readCompletionDelegate and it seemed like a race condition that this member would be set after calling StartReceiveMessage.  Albeit an unlikely race condition.  I will remove this when I separate my pull requests; it is not prerequisite functionality for what I'm trying to accomplish.